contracts/rosca.sol

```diff
@@ -1,72 +1,266 @@
 pragma solidity ^0.4.4;
 
+/*
+*  Possible missing functions
+* - integrity check? to perform check and make sure certain values are available and correct
+*
+*/
 contract rosca {
-    address WETRUST = 0;  // constant
-    
-    event LogParticipantRegistered(string uniqName);
-    event LogContributionMade(string uniqName);
-    event LogBidPlaced(string uniqName);
-    event LogRoundFundsRelease(string winnerUniqName, uint amountInWei);
-    
-    
-    enum Period { Weekly, Monthly }
-    
+    address constant WETRUST = 0x0;
+
+    enum Period { Weekly, Monthly } // are we only allwing 2 options?
+
+    event LogParticipantRegistered(address user);
+    event LogContributionMade(address user, uint amount);
+    event LogNewLowestBid(uint bid,address winnerAddress);
+    event LogRoundFundsRelease(address winnerUniqName, uint amountInWei);
+
+
+    //added
+    event LogFundsWithdrawal(address user, uint amount,address destination);
+    event LogStartOfPeriod(uint currentRound);
+
+    //state variables
+    address foreman;
+    uint roundSum;
+    uint numRounds;
+    uint currentEpoch;
+    uint numEpochs;
+    int8 minParticipants;
```

Comment:
nit: just to prevent small bugs if someone decides to run a bigger ROSCA, perhaps we can change this to int16? anyway this won't use up more space because the EVM bunches together variables to 256-bit chunks